actions-server/src/handlers/joinGame.ts

```diff
@@ -0,0 +1,86 @@
+import { Request, Response } from "express"
+import { DocumentNode } from "graphql"
+import jwt from "jsonwebtoken"
+import fetch from "node-fetch"
+import {
+  InsertPlayerForGameDocument,
+  InsertPlayerForGameMutationVariables,
+  LookupPlayerForGameDocument,
+  LookupPlayerForGameQueryVariables,
+} from "../../../app/src/generated/graphql"
+
+// execute the parent operation in Hasura
+const execute = async (query: DocumentNode, variables: object) => {
+  const fetchResponse = await fetch(process.env.HASURA_ENDPOINT, {
+    method: "POST",
+    body: JSON.stringify({
+      query,
+      variables,
+    }),
+  })
+  const data = await fetchResponse.json()
+  console.log("DEBUG: ", data)
+  return data
+}
+
+// Request Handler
+const handler = async (req: Request, res: Response) => {
+  // get request input
+  const { gameId, clientUuid } = req.body.input
+
+  // execute the Hasura operation(s)
+  let playerId
+  const { data: lookupData, errors } = await execute(
+    LookupPlayerForGameDocument,
+    {
+      gameId,
+      clientUuid,
+    } as LookupPlayerForGameQueryVariables
+  )
+  if (errors) {
+    return res.status(400).json(errors[0])
+  }
+  if (lookupData.players[0]) {
+    // already joined game
+    playerId = lookupData.players[0].id
+  } else {
+    // new player for game
+    const { data: insertData, errors } = await execute(
+      InsertPlayerForGameDocument,
+      {
+        gameId,
+        clientUuid,
+      } as InsertPlayerForGameMutationVariables
```

Comment:
typing!

Reply:
Nice!

Casting probably isn't getting us much here though; looking through this code, I think `execute()` might benefit from some generics?

Reply:
I agree, it's manual... ideally we could pass in the generic so we could do something like... InsertPlayerForGameMutation.Document, InsertPlayerForGameMutation.Variables

some sort of namespace - ive seen it before

might be related to https://graphql-code-generator.com/docs/generated-config/typescript-compatibility#nonamespaces-boolean-default-value-false

need to play around w/it more

Reply:
Addressed in 2878fd1ac0088ac5fa8612a29390025b14a50477.